src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITValue.cpp

```diff
@@ -169,6 +195,66 @@ JITValuePointer LLVMJITValue::div(JITValue& rh) {
       getValueTypeTag(), parent_function_, ans, "div", false);
 }
 
+JITValuePointer LLVMJITValue::div_with_error_check(JITValue& rh) {
+  LLVMJITValue& llvm_rh = static_cast<LLVMJITValue&>(rh);
+  checkOprandsType(this->getValueTypeTag(), rh.getValueTypeTag(), "div");
+
+  llvm::Type* rh_llvm_type =
+      getLLVMType(llvm_rh.getValueTypeTag(), parent_function_.getLLVMContext());
+
+  auto zero_const = rh_llvm_type->isIntegerTy()
+                        ? llvm::ConstantInt::get(rh_llvm_type, 0, true)
+                        : llvm::ConstantFP::get(rh_llvm_type, 0.);
+
+  llvm::BasicBlock* div_ok{nullptr};
+  llvm::BasicBlock* div_zero{nullptr};
+
+  div_ok = llvm::BasicBlock::Create(
+      parent_function_.getLLVMContext(), "div_ok", &parent_function_.func_);
+  div_zero = llvm::BasicBlock::Create(
+      parent_function_.getLLVMContext(), "div_zero", &parent_function_.func_);
+
+  getFunctionBuilder(parent_function_)
+      .CreateCondBr(
+          zero_const->getType()->isFloatingPointTy()
+              ? getFunctionBuilder(parent_function_)
+                    .CreateFCmp(llvm::FCmpInst::FCMP_ONE, llvm_rh.load(), zero_const)
+              : getFunctionBuilder(parent_function_)
+                    .CreateICmp(llvm::ICmpInst::ICMP_NE, llvm_rh.load(), zero_const),
+          div_ok,
+          div_zero);
+
+  getFunctionBuilder(parent_function_).SetInsertPoint(div_ok);
+
+  llvm::Value* ans = nullptr;
+  switch (getValueTypeTag()) {
+    case JITTypeTag::INT8:
+    case JITTypeTag::INT16:
+    case JITTypeTag::INT32:
+    case JITTypeTag::INT64:
```

Comment:
better to reuse and call `div(JITValue& rh)` method and simply add if-else cmp branch？